pkg/apis/upgrade.cattle.io/v1/types.go

```diff
@@ -73,13 +73,15 @@ type VolumeSpec struct {
 
 // DrainSpec encapsulates `kubectl drain` parameters minus node/pod selectors.
 type DrainSpec struct {
-	Timeout                  *time.Duration `json:"timeout,omitempty"`
-	GracePeriod              *int32         `json:"gracePeriod,omitempty"`
-	DeleteLocalData          *bool          `json:"deleteLocalData,omitempty"`
-	IgnoreDaemonSets         *bool          `json:"ignoreDaemonSets,omitempty"`
-	Force                    bool           `json:"force,omitempty"`
-	DisableEviction          bool           `json:"disableEviction,omitempty"`
-	SkipWaitForDeleteTimeout int            `json:"skipWaitForDeleteTimeout,omitempty"`
+	Timeout                  *time.Duration        `json:"timeout,omitempty"`
+	GracePeriod              *int32                `json:"gracePeriod,omitempty"`
+	DeleteLocalData          *bool                 `json:"deleteLocalData,omitempty'"`
+	DeleteEmptydirData       *bool                 `json:"deleteEmptydirData,omitempty"`
```

Comment:
Looks like `DeleteEmptydirData` is essentially "alias" for `DeleteLocalData`. I understand the `DeleteEmptydirData` is more appropriate name for what the setting is for now and we can't remove/rename field for backward compatibility purposes.

But adding this field adds the potential to cause confusion for SUC user where they can set one of `DeleteEmptydirData` and `DeleteLocalData` to `true` and the other to `false` and expecting something that SUC doesn't actually support.

So my recommendation is to leave things be "as is" and not add new `DeleteEmptydirData` field.

@Oats87 @brandond , WDYT?

Reply:
The flag has been renamed since Kubernetes 1.20 and the old flag is totally gone from the upstream docs. I'd prefer to start transitioning over to the proper name rather than just stay with something that is going to be confusing for anyone approaching this with no knowledge of what this actually maps to on modern versions of Kubernetes. I do agree that the current behavior of just silently allowing conflicting settings could lead to confusion. Let me take a shot at addressing that.